sympy/functions/elementary/hyperbolic.py

```diff
@@ -1558,6 +1603,16 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
     def _eval_rewrite_as_log(self, x, **kwargs):
         return (log(1 + 1/x) - log(1 - 1/x)) / 2
 
+    def _eval_rewrite_as_atanh(self, x, **kwargs):
+        return atanh(1/x)
+
+    def _eval_rewrite_as_asinh(self, x, **kwargs):
+        from sympy.functions.elementary.complexes import arg, Abs
+        return Piecewise((S.Half*asinh(2*x/(1 - x**2)) - pi*x/2*sqrt(-1/x**2),
+                          Abs(x) < 1),
+                         (-S.Half*asinh(2*x/(1 - x**2)), Abs(x) > 1),
+                         (S.Infinity*arg(x), True))
```

Comment:
This is `acoth(x).rewrite(asinh)`. The final case in the Piecewise seems incorrect for `x=I`:
```python
In [1]: acoth(x).subs(x, I)
Out[1]: 
-ⅈ⋅π 
─────
  4  

In [2]: acoth(x).rewrite(asinh).subs(x, I)
Out[2]: ∞
```